src/USER-HMA/compute_hma.cpp

```diff
@@ -0,0 +1,473 @@
+
+  if (fix->restart_reset) fix->restart_reset = 0;   
+  else {
+    double **xoriginal = fix->astore;            
+    double **x = atom->x;
+    imageint *image = atom->image;
+    int nlocal = atom->nlocal;
+
+    for (int i = 0; i < nlocal; i++)
+      domain->unmap(x[i],image[i],xoriginal[i]);          
+  }
+
+  vector_flag = 1;
+  extvector = -1;
+  comm_forward = 0; // 3 if 2nd derivative needed
+
+  computeU = computeP = computeCv = -1;
+  returnAnharmonic = 0;
+  std::vector<int> extvec;
+  for (int iarg=4; iarg<narg; iarg++) {
+    if (!strcasecmp(arg[iarg], "u")) {
+      computeU = extvec.size();
+      extvec.push_back(1);
+    }
+    else if (!strcasecmp(arg[iarg], "p")) {
+      if (iarg+2 > narg) error->all(FLERR,"Illegal fix hma command");
+      computeP = extvec.size();
+      deltaPcap = force->numeric(FLERR, arg[iarg+1]);
+      extvec.push_back(0);
+      iarg++;
+    }
+    else if (!strcasecmp(arg[iarg], "cv")) {
+      computeCv = extvec.size();
+      comm_forward = 3;
+      extvec.push_back(1);
+    }
+    else if (!strcasecmp(arg[iarg], "anharmonic")) {
+      // the first time we're called, we'll grab lattice pressure and energy
+      returnAnharmonic = -1;
+    }
+    else {
+      error->all(FLERR,"Illegal fix hma command");
+    }
+  }
+
+  if (extvec.size() == 0) {
+    error->all(FLERR,"Illegal fix hma command");
+  }
+  size_vector = extvec.size();
+  memory->create(vector, size_vector, "hma::vector");
+  extlist = new int[size_vector];
+  for (int i=0; i<size_vector; i++) {
+    extlist[i] = extvec[i];
+  }
+
+  if (computeU>-1 || computeCv>-1) {
+   peflag = 1;                             
+  }
+  if (computeP>-1) {
+    pressflag = 1;
+  }
+
+  nmax = 0;
+}
+
+/* ---------------------------------------------------------------------- */
+
+ComputeHMA::~ComputeHMA()
+{
+  // check nfix in case all fixes have already been deleted
+  if (modify->nfix) modify->delete_fix(id_fix);
+
+  delete [] id_fix;
+  delete [] id_temp;
+  delete [] extlist;
+  memory->destroy(vector);
+  memory->destroy(deltaR);
+}
+
+/* ---------------------------------------------------------------------- */
+
+void ComputeHMA::init() {
+  int irequest = neighbor->request(this,instance_me);
+  neighbor->requests[irequest]->pair = 0; 
+  neighbor->requests[irequest]->compute = 1; 
+  neighbor->requests[irequest]->occasional = 1; 
+}
+
+void ComputeHMA::init_list(int id, NeighList *ptr)
+{
+  list = ptr;
+}
+
+void ComputeHMA::setup()
+{
+  int dummy=0;
+  int ifix = modify->find_fix(id_temp);
+  if (ifix < 0) error->all(FLERR,"Could not find compute hma temperature ID");
+  double * temperat = (double *) modify->fix[ifix]->extract("t_target",dummy);      
+  if (temperat==NULL) error->all(FLERR,"Could not find compute hma temperature ID");
+  finaltemp = * temperat;       
+
+
+  // set fix which stores original atom coords
+
+  int ifix2 = modify->find_fix(id_fix);
+  if (ifix2 < 0) error->all(FLERR,"Could not find compute hma ID");
```

Comment:
the code looks for a fix, but the error message talks about a compute. am i missing something here?